mlflow/pyfunc/model.py

```diff
@@ -73,16 +74,57 @@ def _log_warning_if_params_not_in_predict_signature(logger, params):
         )
 
 
-class PythonModel:
+class _PythonModelMetaclass(ABCMeta):
+    def __call__(cls, *args, **kwargs):
+        cls._warn_on_setting_model_in_init()
+        return super().__call__(*args, **kwargs)
+
+    def _warn_on_setting_model_in_init(cls):
+        if not cls.__bases__ == (PythonModel,):
+            return
+
+        import click
+
+        message = (
+            click.style(
+                "It looks like you're trying to save a model as an instance attribute. ",
+                fg="yellow",
+            )
+            + click.style("This is not recommended ", fg="yellow", bold=True)
+            + click.style(
+                "as it can cause problems with model serialization, "
+                "especially for large models. ",
+                fg="yellow",
+            )
+            + click.style("Please use the `artifacts` parameter", fg="yellow", bold=True)
+            + click.style(
+                ", and load your external model in the `load_context()` method instead.",
+                fg="yellow",
+            )
+        )
```

Comment:
Does the color work on Databricks notebook?

Reply:
<img width="834" alt="Screenshot 2023-11-17 at 3 27 44 PM" src="https://github.com/mlflow/mlflow/assets/148037680/bd29ffbc-55ed-40ff-ada0-349b1e4492c2">

it does but it's not yellow

Reply:
Where does red come from?

Reply:
I'm just guessing here but I think the ANSI escape codes function as a color ID and the actual color depends on whatever is rendering it

Reply:
Do we really need colored text?

Reply:
We have custom CSS rules in Databricks. Not even shell command overrides will work. Best not to use colors. We can use bold / italics / typeface changes, but color is going to get overridden at DOM loading, regardless of what you're instructing the flexbox div to display.